src/compat.jl

```diff
@@ -1,3 +1,23 @@
 if VERSION < v"1.2"
     Base.getproperty(x::Tuple, f::Int) = getfield(x, f)
 end
+
+if VERSION < v"1.1"
+    # Note: these are actually *better* than the ones in julia 1.1, 1.2, 1.3,and 1.4
```

Comment:
Should we maybe just roll our own versions of these on all Julia versions?

Reply:
no, because it will trigger a lot of recompilation, for not all that much gain.
It never effects us right now anyway.